Nodejs/Product/Npm/PackageSet.cs

```diff
@@ -0,0 +1,110 @@
+//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.NodejsTools.Npm {
+    /// <summary>
+    /// Immutable collection of packages.
+    /// </summary>
+    public class PackageSet : IEnumerable<IPackage> {
+        /// <summary>
+        /// Different between a left package set and right package set.
+        /// </summary>
+        public class Diff {
```

Comment:
This entire class strikes me as pretty opaque. 
```cs
List<string> GetMissingTypings(List<string> topLevelPackages, List<string> typingsFiles, List<string> typingsCache)
```
is more descriptive and gets to the heart of what we're actually trying to do here.